nntrainer/layers/embedding.cpp

```diff
@@ -104,7 +105,7 @@ void EmbeddingLayer::forwarding(RunLayerContext &context, bool training) {
       }
 
       Tensor cur_weight =
-        weight.getSharedDataTensor(out_tensor_dim, out_dim * embed_idx);
+        weight.getSharedDataTensor(weight_dim, out_dim * embed_idx);
```

Comment:
I think embedding weight can differ from the output tensor. So it is correct to handle both of them differently as you pointed out. But check once again if the ```out_tensor.copyData(cur_weight)```is working properly for different tensors.  for example, ```FP16.copyData(Qint8)```